File: solutions/3508-number-of-bit-changes-to-make-two-integers-equal/solution.js

var minChanges = function(n, k) {
    let count=0;

    while(n>0 || k>0){
        let nBit=n%2;
        let kBit=k%2;
        if(nBit==0 && kBit==1){
            return -1;
        }
        if(nBit==1 && kBit==0){
            count +=1
        }
        n=Math.floor(n / 2);
        k=Math.floor(k / 2);
        
    }
    return count; 
};

File: solutions/3508-number-of-bit-changes-to-make-two-integers-equal/solution.java
class Solution {
    public int minChanges(int n, int k) {
        int count=0;
        while(n > 0 || k>0){
            int nBit= n%2;
            int kBit=k%2;
            if(nBit == 1 && kBit == 0){
                count += 1;
            }
            if(nBit == 0 && kBit == 1){
                return -1;
            }

            n=  (int)Math.floor(n/2);
            k= (int)Math.floor(k/2);
        }
        return count;
    }
}
